src/Profiles.ts

```diff
@@ -336,6 +347,218 @@ export class Profiles {
+                const findNode = ses.label.substring(1, ses.label.indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    jobsProvider.removeFavorite(ses);
+                    ses.dirty = true;
+                    jobsProvider.refresh();
+                }
+            });
+
+            // Delete from Jobs Tree
+            jobsProvider.mSessionNodes.forEach((jobNode) => {
+                if (jobNode.getProfileName() === deleteLabel) {
+                    jobsProvider.deleteSession(jobNode);
+                    jobNode.dirty = true;
+                    jobsProvider.refresh();
+                }
+            });
+
+            // Delete from Data Set Sessions list
+            const dsSetting: any = {...vscode.workspace.getConfiguration().get(this.dsSchema)};
+            let sessDS: string[] = dsSetting.sessions;
+            let faveDS: string[] = dsSetting.favorites;
+            sessDS = sessDS.filter( (element) => {
+                return element.trim() !== deleteLabel;
+            });
+            faveDS = faveDS.filter( (element) => {
+                return element.substring(1, element.indexOf("]")).trim() !== deleteLabel;
+            });
+            dsSetting.sessions = sessDS;
+            dsSetting.favorites = faveDS;
+            await vscode.workspace.getConfiguration().update(this.dsSchema, dsSetting, vscode.ConfigurationTarget.Global);
+
+            // Delete from USS Sessions list
+            const ussSetting: any = {...vscode.workspace.getConfiguration().get(this.ussSchema)};
+            let sessUSS: string[] = ussSetting.sessions;
+            let faveUSS: string[] = ussSetting.favorites;
+            sessUSS = sessUSS.filter( (element) => {
+                return element.trim() !== deleteLabel;
+            });
+            faveUSS = faveUSS.filter( (element) => {
+                return element.substring(1, element.indexOf("]")).trim() !== deleteLabel;
+            });
+            ussSetting.sessions = sessUSS;
+            ussSetting.favorites = faveUSS;
+            await vscode.workspace.getConfiguration().update(this.ussSchema, ussSetting, vscode.ConfigurationTarget.Global);
+
+            // Delete from Jobs Sessions list
+            const jobsSetting: any = {...vscode.workspace.getConfiguration().get(this.jobsSchema)};
+            let sessJobs: string[] = jobsSetting.sessions;
+            let faveJobs: string[] = jobsSetting.favorites;
+            sessJobs = sessJobs.filter( (element) => {
+                return element.trim() !== deleteLabel;
+            });
+            faveJobs = faveJobs.filter( (element) => {
+                return element.substring(1, element.indexOf("]")).trim() !== deleteLabel;
+            });
+            jobsSetting.sessions = sessJobs;
+            jobsSetting.favorites = faveJobs;
+            await vscode.workspace.getConfiguration().update(this.jobsSchema, jobsSetting, vscode.ConfigurationTarget.Global);
```

Comment:
There seems to be a lot of knowledge duplication here. How about this:

```suggestion
    // Delete from Session List
    const isNotDeletedSession = (element) => element.trim() !== deleteLabel;
    const isNotDeletedFavorite = (element) =>
      element.substring(1, element.indexOf("]")).trim() !== deleteLabel;

    const deleteFromSessionList = async (schema) => {
      const setting: {
        settings: string[];
        favorites: string[];
      } = vscode.workspace.getConfiguration().get(schema);
      await vscode.workspace.getConfiguration().update(
        schema,
        {
          ...setting,
          session: setting.favorites.filter(isNotDeletedSession),
          favorites: setting.favorites.filter(isNotDeletedFavorite),
        },
        vscode.ConfigurationTarget.Global
      );
    };

    deleteFromSessionList(this.dsSchema);
    deleteFromSessionList(this.ussSchema);
    deleteFromSessionList(this.jobsSchema);
```